src/e2eTest/utils/actions/custom-actions/createCase.action.ts

```
@@ -252,6 +255,37 @@ export class CreateCaseAction implements IAction {
     await performAction('clickButton', 'Continue');
   }
 
+  private async selectYourPossessionGrounds(possessionGrounds: actionData) {
+    const grounds = possessionGrounds as {
+      mandatory?: string[];
+      mandatoryAccommodation?: string[];
+      discretionary?: string[];
+      discretionaryAccommodation?: string[];
+    };
+    if (grounds.discretionary) {
+      await performAction('check', grounds.discretionary);
+    }
+    if (grounds.mandatory) {
+      await performAction('check', grounds.mandatory);
+    }
+    if (grounds.mandatoryAccommodation) {
+      await performAction('check', grounds.mandatoryAccommodation);
+    }
+    if (grounds.discretionaryAccommodation) {
+      await performAction('check', grounds.discretionaryAccommodation);
+    }
+    await performAction('clickButton', 'Continue');
+  }
+
+  private async enterReasonForPossession(caseData: actionData) {
```

Would it be better to name this parameter something like `reasons` rather than case data? Also could you make the type into a string[]?

With regards to the Array check below, if it is not an array, doesn't that mean that there is an issue with the test configuration? If so, should it fail fast?

---

Thanks @scottstewart-sl , As discussed I have renamed the parameter and also handled the exception/not an array scenario